data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/configuration/EntryConfig.java

```diff
@@ -30,7 +44,48 @@ public String getTarget() {
         return target;
     }
 
-    public List<String> getFields() {
-        return fields;
+
+    public List<GeoIPField> getFields() {
```

Comment:
I think you could consider moving this to an abstraction class. But, I don't feel as strongly about that as with the databases. These are more closely related to the configuration than the databases.